### PR TITLE
switch from Ownable to Roles for more granular access control

### PR DESCRIPTION
**Pull request by kentkolze**
## What?

Switch from Ownable to using Roles. Allow Admin and Operator roles

## Why?

Allows more granular access control, specifically we want to enable a blessed operator to handle transference

### CODE REVIEW

File: chain/src/CapTable.sol [hunk 1]
```diff
@@ -290,4 +296,35 @@ contract CapTable is Ownable {
     function _safeNow() internal view returns (uint40) {
         return uint40(block.timestamp);
     }
+
+    /* RBAC */
+
+    modifier isOperator() {
+        // TODO: should admins be operators?
+        require(operators.has(msg.sender) || admins.has(msg.sender), "Does not have operator role");
```
Comment (victormimo) [inline chain/src/CapTable.sol:304]: I think yes, admins should be a subset of operators

File: chain/src/CapTable.sol [hunk 2]
```diff
@@ -290,4 +296,35 @@ contract CapTable is Ownable {
...
+    }
+
+    modifier isAdmin() {
+        require(admins.has(msg.sender), "Does not have admin role");
+        _;
+    }
+
+    // TODO: kent -- is there a more elegant way for the below? I hate boilerplate code bloat
+    function addAdmin(address memory addr) external isAdmin {
+        admins.add(addr);
```
Comment (victormimo) [inline chain/src/CapTable.sol:315]: I can't think of another way top of mind but it definitely adds code bloat.
Should we create a Roles internal library that contains these functions?

Comment (kentkolze) [reply chain/src/CapTable.sol:313]: as discussed, we are going to keep as is

File: chain/src/CapTable.sol [hunk 3]
```diff
@@ -1,15 +1,15 @@
 // SPDX-License-Identifier: MIT
 pragma solidity ^0.8.20;
 
-import "openzeppelin-contracts/contracts/access/Ownable.sol";
+import "openzeppelin-contracts/contracts/access/Roles.sol";
```
Comment (victormimo) [inline chain/src/CapTable.sol:4]: This import should work.. here's openzeppelin-solidity being used?

**Pull request closed without merging**